server/routes.ts

```diff
@@ -434,11 +434,20 @@ export class Router {
       }),
     );
 
-    app.get(`/device/profile/:id`, proxy(proxyUrl, {
+    app.get(`/device/profile/:id`,
+    async (req, res, next) => {
+      logger.debug(`Received API call to get device profile`);
+      const apiKey = await containerAPI.getDeviceSdkInstance().getToken().catch((err) => {
+          logger.error(`Received error while fetching api key in location search with error: ${err}`);
```

Comment:
the error message is incorrect, it is device profile, not location search

Reply:
Done